dlc/src/main/scala/org/bitcoins/dlc/DLCClient.scala

```diff
@@ -238,6 +239,39 @@ case class DLCClient(
     sigsMapF.map(FundingSignatures.apply)
   }
 
+  def verifyRemoteFundingSigs(remoteSigs: FundingSignatures): Boolean = {
+    val remoteTweak = if (isInitiator) {
+      fundingUtxos.length
+    } else {
+      0
+    }
+
+    val psbt = PSBT.fromUnsignedTx(createUnsignedFundingTransaction)
+
+    remoteSigs.zipWithIndex
+      .foldLeft(true) {
+        case (ret, ((outPoint, sigs), index)) =>
+          require(psbt.transaction.inputs(index).previousOutput == outPoint,
+                  "Adding signature for incorrect input")
+
+          val idx = index + remoteTweak
+
+          psbt
+            .addWitnessUTXOToInput(remoteFundingInputs(index).output, idx)
+            .addSignatures(sigs, idx)
+            .inputMaps(idx)
+            .finalize(psbt.transaction.inputs(idx)) match {
+            case Success(inputMap) =>
+              val newInputMaps = psbt.inputMaps.updated(idx, inputMap)
+              val finalized = psbt.copy(inputMaps = newInputMaps)
```

Comment:
I think this should just be an added method to `PSBT` called `finalizeInput` which takes an index and returns a new PSBT